File: Python/338_Counting_Bits.py
class Solution:
    def countBits(self, num: int) -> List[int]:
        offset = 1
        ans = [0]*(num+1)
        
        for i in range(1,num+1):
            if offset*2 == i:
                offset = i
            ans[i] = ans[i-offset] + 1
        return ans

File: C++/338_Counting_Bits.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        int offset = 1;
        vector<int> res(num+1,0);

        for(int i=1;i<=num;i++)
        {
            if(offset*2==i)
            {
                offset = i;
            }
            res[i] = res[i-offset]+1;
        }
        return res;
    }
};
